### PR TITLE
support positional `--mp` parameter for `forge test`

### PR DESCRIPTION
**Pull request by zhiqiangxu**
This PR tries to fix [this issue](https://github.com/foundry-rs/foundry/issues/8742), users can now simply type `forge test path_to_sol` to run test.

### CODE REVIEW

File: crates/forge/bin/cmd/test/mod.rs [hunk 1]
```diff
@@ -634,6 +639,14 @@ impl TestArgs {
         if self.rerun {
             filter.test_pattern = last_run_failures(config);
         }
+        if filter.path_pattern.is_some() {
+            if self.path.is_some() {
+                println!("Can not supply both --match-path and |path|");
+                std::process::exit(0);
```
Comment (zerosnacks) [inline crates/forge/bin/cmd/test/mod.rs:645]: It is better to use `panic!` here, exit code `0` indicates success

Comment (zhiqiangxu) [reply crates/forge/bin/cmd/test/mod.rs]: Done.

**Comment by zhiqiangxu** [comment]
@zerosnacks Do you have any additional feedback or suggestions for this PR?

**Review by klkvr — APPROVED** [review]

**Review by zerosnacks — APPROVED** [review]
LGTM, thanks @zhiqiangxu !

**Pull request merged**